src/framework/auth/services/auth.service.ts

```diff
@@ -34,12 +47,29 @@ export class NbAuthService {
   }
 
   /**
-   * Returns true if auth token is presented in the token storage
+   * Returns true if valid auth token is present in the token storage.
+   * If not, calls the strategy refreshToken, and returns true if it get a new valid access token, false otherwise
    * @returns {Observable<any>}
    */
   isAuthenticated(): Observable<boolean> {
     return this.getToken()
-      .pipe(map((token: NbAuthToken) => token.isValid()));
+      .pipe(
+        switchMap(token => {
+        if (!token.isValid() && (token.getOwnerStrategyName())) {
+          return this.refreshToken(token.getOwnerStrategyName(), token)
+            .pipe(
+              switchMap(res => {
+                if (res.isSuccess()) {
+                  return observableOf(res.getToken().isValid());
```

Comment:
and then if isSuccess === true, we should call the isAuthenticated method again without refreshing a token, just to make sure that the whole process behind went well.